server/storage.js

```diff
@@ -82,4 +83,24 @@ app.post('/delete', wrap(async (req, res) => {
   res.redirect('/storage/list');
 }));
 
+/**
+ * Duplicate the mail in redis db
+ */
+app.post('/duplicate', wrap(async (req, res) => {
+  var content_json = await fs.readFile(path.join('./emails/', req.body.id+'.json'), {encoding: 'utf-8'});
+
+  var metadata = JSON.parse(content_json).metadata;
+  metadata.name = req.body.email_name; //new name
+
+  var uuid = newUuid();
+  await redis.lpushAsync(`mosaico:${req.user}:emails`, uuid);
+  await fs.writeFile(`./emails/${uuid}.html`, fs.readFile(path.join('./emails/', req.body.id+'.html'), {encoding: 'utf-8'}));
```

Comment:
I guess this line does not work. fs.readFile is a promise, not the content of the file. You should use .pipe()